docs(astro): Add multi-framework example for $signInStore and $signUpStore (#1435)
clerk/clerk-docs: docs/references/astro/sign-up-store.mdx

# `$signUpStore`

> The `$signUpStore` store provides a convenient way to access the [`SignUp`](https://clerk.com/docs/references/javascript/sign-up/sign-up) object, which allows you to check the current state of a sign-up. This is also useful for creating a custom sign-up flow.

Reviewer: ```suggestion
The `$signUpStore` store provides a convenient way to access the [`SignUp`][`SignUp`](/docs/references/javascript/sign-up/sign-up){{ target: '_blank' }} object, which allows you to check the current state of a sign-up. This is also useful for creating a custom sign-up flow.
```

> The `$signUpStore` store provides a convenient way to access the \[`SignUp`][`SignUp`](/docs/references/javascript/sign-up/sign-up){{ target: '_blank' }} object, which allows you to check the current state of a sign-up. This is also useful for creating a custom sign-up flow.

Reviewer 2: ```suggestion
The `$signUpStore` store provides a convenient way to access the [`SignUp`](/docs/references/javascript/sign-up/sign-up){{ target: '_blank' }} object, which allows you to check the current state of a sign-up. This is also useful for creating a custom sign-up flow.
```

## How to use the `$signUpStore` store

### Check the current state of a sign-up

The following example demonstrates how to use the `$signUpStore` store to access the `SignUp` object and check the current state of a sign-up.

<CodeBlockTabs options={['React', 'Vue', 'Svelte']}>

>   ```tsx {{ prettier: false, filename: 'sign-up-step.tsx' }}

Reviewer: Same question about formatting the React code block

```suggestion
  ```tsx {{ filename: 'sign-up-step.tsx' }}
```

    const signUp = useStore($signUpStore);

    if (signUp === undefined) {
      // Add logic to handle loading state
      return null;
    }

    return <div>The current sign-up attempt status is {signUp.status}.</div>;
  }
  ```

  ```vue {{ prettier: false, filename: 'sign-up-step.vue' }}
  <script setup>
  import { useStore } from '@nanostores/vue';
  import { $signUpStore } from '@clerk/astro/client';

  const signUp = useStore($signUpStore);
  </script>

  <template>
    <div v-if="signUp === undefined">
      <!-- Add logic to handle loading state -->
    </div>
    <div v-else>
      <div>The current sign-up attempt status is {{ signUp.status }}.</div>
    </div>
  </template>
  ```

  ```svelte {{ prettier: false, filename: 'sign-up-step.svelte' }}
  <script>
  // The $ prefix is reserved in Svelte for its own reactivity system.
  // Alias the imports to avoid conflicts.
  import { $signUpStore as signUp } from '@clerk/astro/client';
  </script>

  {#if $signUp === undefined}
      <!-- Add logic to handle loading state -->
  {:else}
      <div>The current sign-up attempt status is {$signUp.status}.</div>
  {/if}
  ```
</CodeBlockTabs>

> The possible values for the `status` property of the `SignUp` resource are listed [here](https://clerk.com/docs/references/javascript/sign-up/sign-up#properties).

Reviewer: ```suggestion
The possible values for the `status` property of the `SignUp` resource are listed [here](/docs/references/javascript/sign-up/sign-up#properties){{ target: '_blank' }}.
```

### Create a custom sign-up flow

> The `$signUpStore` store can also be used to build fully custom sign-up flows, if Clerk's pre-built components don't meet your specific needs or if you require more control over the authentication flow. Different sign-up flows include email and password, email and phone codes, email links, and multifactor (MFA). To learn more about using the `$signUpStore` store to create custom flows, check out the [custom flow guides](/docs/custom-flows/overview)

Reviewer: ```suggestion
The `$signUpStore` store can also be used to build fully custom sign-up flows, if Clerk's pre-built components don't meet your specific needs or if you require more control over the authentication flow. Different sign-up flows include email and password, email and phone codes, email links, and multifactor (MFA). To learn more about using the `$signUpStore` store to create custom flows, check out the [custom flow guides.](/docs/custom-flows/overview)
```
